tests/components/zwave_js/test_device_action.py

```diff
@@ -0,0 +1,412 @@
+        },
+    )
+    assert capabilities and "extra_fields" in capabilities
+
+    assert voluptuous_serialize.convert(
+        capabilities["extra_fields"], custom_serializer=cv.custom_serializer
+    ) == [
+        {
+            "name": "value",
+            "required": True,
+            "valueMin": 0,
+            "valueMax": 124,
+        }
+    ]
+
+    # Test undefined type param
+    capabilities = await device_action.async_get_action_capabilities(
+        hass,
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "device_id": device.id,
+            "type": "set_config_parameter",
+            "parameter": 2,
+            "bitmask": None,
+            "subtype": f"{node.node_id}-112-0-2 (HVAC Settings)",
+        },
+    )
+    assert not capabilities
+
+
+async def test_get_action_capabilities_lock_triggers(
+    hass: HomeAssistant,
+    client: Client,
+    lock_schlage_be469: Node,
+    integration: ConfigEntry,
+):
+    """Test we get the expected action capabilities for lock triggers."""
+    dev_reg = device_registry.async_get(hass)
+    device = device_registry.async_entries_for_config_entry(
+        dev_reg, integration.entry_id
+    )[0]
+
+    # Test clear_lock_usercode
+    capabilities = await device_action.async_get_action_capabilities(
+        hass,
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "device_id": device.id,
+            "entity_id": "lock.touchscreen_deadbolt",
+            "type": "clear_lock_usercode",
+        },
+    )
+    assert capabilities and "extra_fields" in capabilities
+
+    assert voluptuous_serialize.convert(
+        capabilities["extra_fields"], custom_serializer=cv.custom_serializer
+    ) == [{"type": "string", "name": "code_slot", "required": True}]
+
+    # Test set_lock_usercode
+    capabilities = await device_action.async_get_action_capabilities(
+        hass,
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "device_id": device.id,
+            "entity_id": "lock.touchscreen_deadbolt",
+            "type": "set_lock_usercode",
+        },
+    )
+    assert capabilities and "extra_fields" in capabilities
+
+    assert voluptuous_serialize.convert(
+        capabilities["extra_fields"], custom_serializer=cv.custom_serializer
+    ) == [
+        {"type": "string", "name": "code_slot", "required": True},
+        {"type": "string", "name": "usercode", "required": True},
+    ]
+
+
+async def test_failure_scenarios(
+    hass: HomeAssistant,
+    client: Client,
+    hank_binary_switch: Node,
+    integration: ConfigEntry,
+):
+    """Test failure scenarios."""
+    dev_reg = device_registry.async_get(hass)
+    device = device_registry.async_entries_for_config_entry(
+        dev_reg, integration.entry_id
+    )[0]
+
+    with pytest.raises(HomeAssistantError):
+        await device_action.async_call_action_from_config(
```

Comment:
Can we trigger an automation instead that calls the action?

Reply:
Setting up the automation fails on config validation so we don't hit this check. Should we just remove the check then? Is there a scenario where someone would have an invalid `type` when `async_call_action_from_config` is called?

Reply:
Maybe assert that the automation setup failed and assert a log message?

Reply:
That's what I did, it just leaves that one line uncovered. Maybe that's ok?

Reply:
We can keep it like this for now until we possibly find a better way.